### PR TITLE
applications: machine_learning: change pwm_period to improve led…

### PR DESCRIPTION
**Pull request by zycz**
… color blinking

Decrease pwm_period for leds on thingy53 so while blinking
led does not shine in different colour.

Jira: NCSDK-15707

Signed-off-by: Jan Zyczkowski <jan.zyczkowski@nordicsemi.no>

### CODE REVIEW

**Review by pdunaj — CHANGES_REQUESTED** [review]
```
bug: applications: machine_learning
```
remove `bug:` from commit message.

BTW - there seems to be nothing in the code that would do what message suggests

**Comment by zycz** [comment]
> ```
> bug: applications: machine_learning
> ```
> 
> remove `bug:` from commit message.
> 
> BTW - there seems to be nothing in the code that would do what message suggests

Removed bug label.
The change adds pwm nodes to dts overlay so it overwrites pwm_period which was previously defined 20ms in zephyr thingy53_common.dts

File: applications/machine_learning/configuration/thingy53_nrf5340_cpuapp/app.overlay [hunk 1]
```diff
@@ -67,6 +67,24 @@
 		};
 	};
 
+	pwmleds {
```
Comment (MarekPieta) [inline applications/machine_learning/configuration/thingy53_nrf5340_cpuapp/app.overlay:70]: Update `_ns` configuration too

File: applications/machine_learning/configuration/thingy53_nrf5340_cpuapp/app.overlay [hunk 2]
```diff
@@ -67,6 +67,24 @@
...
+		compatible = "pwm-leds";
+		red_led_pwm: led_pwm_0 {
+			pwms = <&pwm0 0 PWM_MSEC(1) PWM_POLARITY_NORMAL>;
+			label = "Red PWM LED";
+		};
+
+		green_led_pwm: led_pwm_1 {
+			pwms = <&pwm0 1 PWM_MSEC(1) PWM_POLARITY_NORMAL>;
+			label = "Green PWM LED";
```
Comment (MarekPieta) [inline applications/machine_learning/configuration/thingy53_nrf5340_cpuapp/app.overlay:79]: Do we need to update labels ?

File: applications/machine_learning/configuration/thingy53_nrf5340_cpuapp/app.overlay [hunk 3]
```diff
@@ -67,6 +67,21 @@
 		};
 	};
 
+	pwmleds {
+		compatible = "pwm-leds";
```
Comment (MarekPieta) [inline applications/machine_learning/configuration/thingy53_nrf5340_cpuapp/app.overlay:71]: Is assigning `compatible` needed?

Comment (MarekPieta) [reply applications/machine_learning/configuration/thingy53_nrf5340_cpuapp/app.overlay:71]: Isn't it already assigned by the board's DTS?

File: doc/nrf/known_issues.rst [hunk 4]
```diff
@@ -1130,6 +1130,13 @@ NCSDK-15675: Possible advertising start failure and module state error in :ref:`
 
   **Workaround:** Manually cherry-pick and apply commit with fix from main (commit hash: ``934a25ac23125758e350b64bca23885486682109``).
 
+.. rst-class:: v2-0-0
+
+NCSDK-15707: Due to changes in pwm drivers default pwm_period has been increased.
+  It causes visual glitches while LED color is updated for LEDs with more than one color channel.
```
Comment (MarekPieta) [inline doc/nrf/known_issues.rst:1136]: Maybe we could provide some context?
```
Update of the first LED channel occurs one PWM period earlier than update of remaining channels. Shorter LED PWM period mitigates the observed issue.
```

Comment (MarekPieta) [reply doc/nrf/known_issues.rst:1136]: Can we also add a short note about the problem in CAF LEDs documentation (I think we need it)

**Review by MarekPieta — CHANGES_REQUESTED** [review]

**Review by greg-fer — CHANGES_REQUESTED** [review]

**Review by MarekPieta — APPROVED** [review]

**Comment by gmarull** [comment]
shouldn't this be set by default in the upstream board?

**Comment by MarekPieta** [comment]
Hi @gmarull
`PWM_MSEC(20)` seems a default period for PWM LEDs of Zephyr boards.

If we use nrfx PWM driver to drive a RGB LED, update of the first LED color occurs one PWM period earlier than update of remaining channels (that causes glitches during e.g. RGB LED blinking). Shorter LED PWM period mitigates the observed issue.  The visual glitch would not affect monochromatic LEDs and cases when color is changed slowly (or cases where only one color channel is used). 

Shorter PWM period leads to increase of power consumption and it may not be needed in some use-cases. I think that if we decide to change the default value, we should do it for all of the boards to use consistent default configuration.

**Review by pdunaj — APPROVED** [review]

**Pull request merged**